c_src/erocksdb_db.cc

```diff
@@ -1886,29 +1903,59 @@ Flush(
         return enif_make_badarg(env);
 
     // parse flush options
-    rocksdb::FlushOptions *opts = new rocksdb::FlushOptions;
-    fold(env, argv[2], parse_flush_option, *opts);
+    rocksdb::FlushOptions opts;
+    fold(env, argv[2], parse_flush_option, opts);
 
     ReferencePtr<ColumnFamilyObject> cf_ptr;
+    std::vector<ReferencePtr<ColumnFamilyObject>> cfs;
     rocksdb::Status status;
-    if(argv[1] == erocksdb::ATOM_DEFAULT_COLUMN_FAMILY)
+    if (argv[1] == erocksdb::ATOM_DEFAULT_COLUMN_FAMILY)
+    {
+        status = db_ptr->m_Db->Flush(opts);
+    }
+    else if (enif_is_list(env, argv[1]))
     {
-        status = db_ptr->m_Db->Flush(*opts);
+        ERL_NIF_TERM head, tail = argv[1];
+        while (enif_get_list_cell(env, tail, &head, &tail))
+        {
+            cfs.emplace_back();
+            if (0 == enif_get_cf(env, head, &cfs.back()))
+            {
+                return enif_make_badarg(env);
+            }
+        }
     }
     else if (enif_get_cf(env, argv[1], &cf_ptr))
     {
-        status = db_ptr->m_Db->Flush(*opts, cf_ptr->m_ColumnFamily);
+        cfs.push_back(cf_ptr);
     }
     else
     {
-        delete opts;
-        opts = NULL;
         return enif_make_badarg(env);
     }
 
-    delete opts;
-    opts = NULL;
-
+    if (cfs.size() > 1)
+    {
+        std::vector<rocksdb::ColumnFamilyHandle*> cf_handles;
+        std::for_each(
```

Comment:
If I read the code correct, this loop simply copies the data created at line 1919-1921 from one vector to another. I wonder if this can be avoided if the the first loop does it.

Reply:
Yes, this is a bit of a waste. Admittedly I have no deep understanding of the mechanics, but `enif_get_cf` works with `ReferencePtr<T :: ErlRefObject>` that does refcounting internally. Which sounds important judging from the surrounding code and weird at the same time because ERTS does this refcounting of NIF resources by itself as far as I know.